dashdotdb/controllers/deploymentvalidation.py

```diff
@@ -1,12 +1,21 @@
 from dashdotdb.services.deploymentvalidation import DeploymentValidationData
+from dashdotdb.controllers.tokens import generate_token, close_token
+from dashdotdb.services import DataTypes
 
 
-def post(cluster, body):
+def post(token, cluster, body):
     dpv = DeploymentValidationData(cluster=cluster)
-    dpv.insert(validation=body)
-    return 'ok'
+    return dpv.insert(token=token, validation=body)
 
 
 def search(cluster, namespace):
-    dpv = DeploymentValidationData(cluster, namespace)
+    dpv = DeploymentValidationData(cluster=cluster, namespace=namespace)
     return dpv.get_deploymentvalidations()
+
+
```

Comment:
So we need 2 new `token` endpoints per data type...

Should we simplify that and have a single endpoint overall?

Like, to get a new token:
```
GET /token?scope='cso'
```

And to finish using it:
```
DELETE /token/<token>
```

That would give us a similar experience to the docker.io tokens (https://docs.docker.com/registry/spec/auth/token/), with one auth endpoint instead of one auth endpoint per data type.

Anyways, I'm fine with the current approach, just wanted to make sure you know my thoughts here :)

Reply:
I was thinking along these lines and trying to find a way to have a common endpoint.  The concern I have, and maybe it isn't justified, is I feel having the scope come in via a parameter seems like making it a magic string that the client would have to know, whereas the name in the path feels less so.

I have wanted a single endpoint for simplicity, and it sounds like there is precedence for doing the scope in a query parameter.  I think that's justification enough.

Reply:
Done

Reply:
well, the token release doesn't delete anything so I thought it could be misleading. But I'm fine if you like it and you don't think it is confusing

Reply:
It doesn't have to be the DELETE method. It can be POST, UPDATE, whatever. But from the client perspective, once the token is released, it can't be reused, so I thought DELETE would be the closest match avaliable to represent that behaviour.

Reply:
I kept this as a POST since this has had a lot of testing as it is.  DELETE to me seems wrong since we aren't actually deleting anything from the server.  I added logic so that you can't re-POST to a closed token as well.

I don't feel strongly about the operation we use here and it doesn't seem like there's one that really fits all that well.  I' for keeping it a POST only because this path has gotten a lot of testing.  If there's a good reason to change it though, I can do that.